test/unit/test_fiat.py

```diff
@@ -604,6 +603,68 @@ def eval_basis(f, pt):
             assert np.allclose(uh, exact, atol=1E-14)
 
 
+@pytest.mark.parametrize('cell', [I, T, S])
+def test_make_bubbles(cell):
+    from FIAT.quadrature_schemes import create_quadrature
+    from FIAT.expansions import polynomial_dimension
+    from FIAT.polynomial_set import make_bubbles, PolynomialSet, ONPolynomialSet
+
+    degree = 10
+    B = make_bubbles(cell, degree)
+
+    # basic tests
+    sd = cell.get_spatial_dimension()
+    assert isinstance(B, PolynomialSet)
+    assert B.degree == degree
+    assert B.get_num_members() == polynomial_dimension(cell, degree - sd - 1)
+
+    # test values on the boundary
+    top = cell.get_topology()
+    points = []
+    for dim in range(len(top)-1):
+        for entity in range(len(top[dim])):
+            points.extend(cell.make_points(dim, entity, degree))
+    values = B.tabulate(points)[(0,) * sd]
+    assert np.allclose(values, 0, atol=1E-12)
+
+    # test linear independence
+    m = B.get_num_members()
+    points = cell.make_points(sd, 0, degree)
+    values = B.tabulate(points)[(0,) * sd]
+    assert values.shape == (m, m)
+    assert np.linalg.matrix_rank(values.T) == m
```

Comment:
Is the default tolerance (which is ~ machine eps) the correct here?  If it passes, that's a strong condition, but could we get any roundoff issue here?